pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/buffer/TransactionBufferClientTest.java

```diff
@@ -330,10 +275,35 @@ public void testTransactionBufferHandlerSemaphore() throws Exception {
         field.setAccessible(true);
         field.set(transactionBufferHandler, new Semaphore(2));
 
-        String topic = "persistent://" + namespace + "/testTransactionBufferLookUp";
-        tbClient.abortTxnOnSubscription(topic + "_abort_sub", "test", 1L, 1L, -1L).get();
+
+        String topic = "persistent://" + namespace + "/testTransactionBufferHandlerSemaphore";
+        String subName = "test";
+
+        admin.topics().createNonPartitionedTopic(topic + "_abort_sub");
+        admin.topics().createSubscription(topic + "_abort_sub", subName, MessageId.earliest);
+
+        admin.topics().createNonPartitionedTopic(topic + "_commit_sub");
+        admin.topics().createSubscription(topic + "_commit_sub", subName, MessageId.earliest);
+
+
+        Awaitility.await().until(() -> {
```

Comment:
why do we have to retry this operation (and the `commit` below) ?

Reply:
waitting pending ack init.